src/mlpack/bindings/cli/add_to_cli11.hpp

```diff
@@ -0,0 +1,192 @@
+        std::get<1>(tuple) = boost::any_cast<std::string>(value);
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
+}
+
+/**
+ * Add an option to CLI11.
+ * 
+ * @param cliName The name of the option to add to CLI11.
+ * @param descr Description string for parameter.
+ * @param desc Options description to add parameter to.
+ */
+template<typename T>
+void AddToCLI11(const std::string& cliName,
+                util::ParamData& param,
+                CLI::App& app,
+                const typename boost::disable_if<std::is_same<T, bool>>::type* = 0,
+                const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0,
+                const typename boost::disable_if<data::HasSerialize<T>>::type* = 0,
+                const typename boost::disable_if<std::is_same<T,
+                   std::tuple<mlpack::data::DatasetInfo, arma::mat>>>::type* = 0)
+{
+  app.add_option_function<T>(cliName.c_str(),
+      [&param](const T& value)
+      {
+        param.value = value;
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
+}
+
+/**
+ * Add a boolean option to CLI11.
+ *
+ * @param cliName The name of the option to add to CLI11.
+ * @param descr Description string for parameter.
+ * @param desc Options description to add parameter to.
+ */
+template<typename T>
+void AddToCLI11(const std::string& cliName,
+                util::ParamData& param,
+                CLI::App& app,
+                const typename boost::enable_if<std::is_same<T, bool>>::type* = 0,
+                const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0,
+                const typename boost::disable_if<data::HasSerialize<T>>::type* = 0,
+                const typename boost::disable_if<std::is_same<T,
+                   std::tuple<mlpack::data::DatasetInfo, arma::mat>>>::type* = 0)
+{
+  app.add_flag_function(cliName.c_str(),
+      [&param](const T& value)
+      {
+        param.value = value;
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
+}
+
+/**
+ * Add an option to CLI11.  This is the function meant to be
+ * used in the CLI function map.
+ *
+ * @param d Parameter data.
+ * @param * (input) Unused void pointer.
+ * @param output Void pointer to options_description object.
+ */
+template<typename T>
+void AddToCLI11(util::ParamData& param,
+                const void* /* input */,
+                void* output)
+{
+  // Cast CLI::App object.
+  CLI::App* app = (CLI::App*) output;
+
+  // Generate the name to be given to CLI11.
+  const std::string mappedName =
+      MapParameterName<typename std::remove_pointer<T>::type>(param.name);
+  std::string cliName = (param.alias != '\0') ?
+      "-" + std::string(1, param.alias) + ",--" + mappedName : "--" + mappedName;
+
+  // Note that we have to add the option as type equal to the mapped type, not
+  // the true type of the option.  
+  AddToCLI11<typename ParameterType<typename std::remove_pointer<T>::type>::type>(
```

Comment:
```suggestion
  AddToCLI11<typename std::remove_pointer<T>::type>(
```
I *think* that this should work.  But in your comment you mentioned that maybe it doesn't?

Reply:
No, it did not compile